keel-igor/src/main/kotlin/com/netflix/spinnaker/keel/services/ArtifactMetadataService.kt

```diff
@@ -0,0 +1,68 @@
+package com.netflix.spinnaker.keel.services
+
+import com.netflix.spinnaker.igor.BuildService
+import com.netflix.spinnaker.keel.api.artifacts.ArtifactMetadata
+import com.netflix.spinnaker.keel.api.artifacts.BuildMetadata
+import com.netflix.spinnaker.keel.api.artifacts.GitMetadata
+import com.netflix.spinnaker.model.Build
+import io.github.resilience4j.retry.annotation.Retry
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Component
+import retrofit2.HttpException
+
+/**
+ * Provides functionality to convert build metadata, which is coming from internal service, to artifact metadata (via igor).
+ */
+@Component
+class ArtifactMetadataService(
+  private val buildService: BuildService
+) {
+
+  @Retry(name = "getArtifactMetadata", fallbackMethod = "getFallback")
```

Comment:
I think it would be good to add a comment here explaining why we need retries. Something like:
```suggestion
  /**
    * Returns additional metadata about the specified build and commit, if available. This call is configured
    * to auto-retry as it's not on a code path where any external retries would happen.
    */
  @Retry(name = "getArtifactMetadata", fallbackMethod = "getFallback")
```